### DISCUSSION TITLE
create inbound fulfillment plan with Fulfillment Inbound API v2024-03-20:

### DISCUSSION THREAD

**Discussion by RNRetailer** [saleweaver/python-amazon-sp-api#1587] [FAQ]
I can easily create a fulfillment plan on v0 of the Fulfillment Inbound API.

However, I can't get it to work on version v2024-03-20.

Do you have a working example of this?

Either fulfillment_inbound.plans or fulfillment_inbound.create_inbound_plan is fine.

Thanks!

**Comment by saleweaver** [comment]
Hi, I don't have an example. Are you passing the version correctly to the client? If you share some code, I might be able to assist

**Comment by RNRetailer** [comment]
```
from download_reports_selling_partner import credentials
from sp_api.api import FulfillmentInbound, FulfillmentInboundVersion
from sp_api.util import throttle_retry

fulfillment_inbound = FulfillmentInbound(
    credentials=credentials,
    version=FulfillmentInboundVersion.LATEST
)

@throttle_retry(tries=10, delay=5, rate=1.3)
def get_fnsku_from_sku(sku):
    res = fulfillment_inbound.create_inbound_plan(body={
        'contactInformation': {'email': 'test@email.com', 'phoneNumber': '5555555555'},

        "destinationMarketplaces": ["ATVPDKIKX0DER"],

        "items": [{
            "labelOwner": "SELLER",
            "msku": sku,
            "prepOwner": "SELLER",
            "quantity": 1
        }],

        "sourceAddress": {
            "addressLine1": "12 main st",
            "addressLine2": "Suite 1000",
            "city": "Tacoma",
            "companyName": "Company Inc",
            "countryCode": "US",
            "email": "test@email.com",
            "name": "John Test",
            "phoneNumber": "5555555555",
            "postalCode": "06989",
            "stateOrProvinceCode": "WA"
        }
    })
```

**Reply by Jitendra9548** [reply]
is there a way to dynamically enter prepOwner selecting between AMAZON/SELLER/NONE

**Comment by RNRetailer** [comment]
sp_api.base.exceptions.SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': "9 validation errors detected: Value '[]' at 'request.destinationMarketplaces' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'request.sourceAddress.phoneNumber' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'request.sourceAddress.city' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'request.sourceAddress.countryCode' failed to satisfy constraint: Member must have length greater than or equal to 2; Value '' at 'request.sourceAddress.countryCode' failed to satisfy constraint: Member must satisfy regular expression pattern: [A-Z]{2}; Value '' at 'request.sourceAddress.postalCode' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'request.sourceAddress.name' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'request.sourceAddress.addressLine1' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '[]' at 'request.items' failed to satisfy constraint: Member must have length greater than or equal to 1", 'details': ''}]

**Comment by RNRetailer** [comment]
Any update? Version 0 will be sunset in December.

Lots of people will need to upgrade.

Thanks

**Comment by RNRetailer** [comment]
@saleweaver

**Comment by RNRetailer** [comment]
@saleweaver

**Comment by RNRetailer** [comment]
please provide an example of this working

the deadline to switch over is in three days

@saleweaver

**Comment by saleweaver** [comment]
Hi, 

I can't check myself, but I'd likely try 

    .create_inbound_plan(**{
      your data
    })

first. The documentation looks like yours should work, but you never know with Amazon.

**Comment by tmrtn** [comment]
Hi,

@saleweaver unfortunately this doesn't work.

@RNRetailer you can try:

`).create_inbound_plan(
    name=name,
    sourceAddress=sourceAddress,
    destinationMarketplaces=[Marketplaces.XX.marketplace_id],
    items=items,
)`
